CMakeLists.txt

```diff
@@ -55,3 +55,21 @@ target_link_libraries(ristnet rist Threads::Threads)
 
 add_executable(rist_cpp main.cpp)
 target_link_libraries(rist_cpp ristnet)
+
+#
+# Build unit tests using GoogleTest
+#
+
+enable_testing()
+
+add_executable(runUnitTests
+        ${CMAKE_CURRENT_SOURCE_DIR}/test/TestRist.cpp
+)
+target_compile_options(runUnitTests PRIVATE -Wall -Wextra -Wno-unused-parameter -Wno-unused-function)
+
+target_include_directories(runUnitTests
+        PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}
+        PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/test
+        PRIVATE ${GTEST_INCLUDE_DIRS})
+
+target_link_libraries(runUnitTests ristnet gtest gtest_main Threads::Threads)
```

Comment:
should use the ```find_package(GTest)``` to simplify the process of adding it to the target to then only use the ```target_link_libraries(GTest::GTest GTest::Main ...)```